app/src/main/java/com/capstone/event_finder/activities/EventDetailsActivity.java

```diff
@@ -127,14 +128,22 @@ private void addEventToBookmarked() {
         String eventId = event.getId();
         String eventCategory = event.getCategory();
         ParseUser currentUser = ParseUser.getCurrentUser();
-        saveBookmark(eventId, eventCategory, currentUser);
+        ParseACL bookmarkAcl = new ParseACL();
+        if (currentUser.get("private").equals(true)) {
+            bookmarkAcl.setReadAccess(ParseUser.getCurrentUser(), true);
+            bookmarkAcl.setWriteAccess(ParseUser.getCurrentUser(), true);
+        } else {
+            bookmarkAcl.setPublicReadAccess(true);
+        }
+        saveBookmark(eventId, eventCategory, currentUser, bookmarkAcl);
```

Comment:
✅ Thanks for giving ACL a try to ensure sever-side security of a user's private bookmarks.